test/utils/manifests.go

```diff
@@ -34,126 +37,69 @@ func GenerateAwsCredentialsSecretFromEnv(secretName, namespace string) *apiv1.Se
-FU/VELIRa6gm8CX5WFbIjgfTzpaMn55WZMHSw/TyPuIZEP+UORzVCoZ9AoGBAPY+
-rD/itlrG7INj+f0sTh7CTVcv6IkP5YasoW5/FHRzjuwK9u6dfyVQ04N7f2KnY36k
-ezdu2w9G/7YA6QdklCPGDWmPrcDDeBaMeNL/TOc3qO1Q6U/jkF/fWuv032ijDCVD
-rz5mM/02tcW8VMrvVgWkLrxUOsr1Ag0yObJtmRzJAoGAWimQH8VQMq4dDC/NOpO0
-SjLki9EQeGE1lsY+4toMvuzQ1bPcuSNaS6nOCtUXYKmx9tx1Kch0oNPDDqLcUnfU
-9ksJySAj8trx9OjkdwhqPumw1eqgfmxGJpnWeLg1JzoBdXBo0s5+DNi6CZHPtUC8
-fNp29AYvGDXlFPQEzvQZjGkCgYBRhtp8pFD/qRCxR66C1eJfaLE2hpQUnQC/H/Sq
-osRg8cmF+PNceSSZdDMzOvYn8YeNbGOnLLq2SilrVs3QNsqdNXtHUdyTD6R4wrVW
-FlSd0N3LBJjabFtmgoqVyJMXD7R7ufcRT8EyuqRf/USNk8QFRiB7FeAJRikRuWlE
-2+hvkQKBgQCE3B4vZKrbUwXuyXnuAiuVFQBSrjVJlpqY38+VucB0UCzUg4BegYCq
-uLVzPuKQRUTVK66PB9KYFsSqGeUsEGpD0pBk3OoqN6Cdio0517IbeSp5tw/JaGnR
-ORj2OTRfjZoPsqrOlUY8A3TVLZc408ECv15PedfW+2TY/2u36XkBFA==
------END RSA PRIVATE KEY-----`
-)
+func ClusterAPIServerAPIServiceObjects(clusterAPINamespace string) (*apiv1.Secret, *apiregistrationv1beta1.APIService, error) {
+	// Copied from the https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/deployer/clusterapiserver.go#L46 (getApiServerCertsForNamespace)
+	name := "clusterapi"
 
-// ClusterAPIServerCertsSecret builds secret object with tls crt and key
-func ClusterAPIServerCertsSecret() *apiv1.Secret {
-	return &apiv1.Secret{
-		Type: "kubernetes.io/tls",
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "cluster-apiserver-certs",
-			Namespace: "default",
-			Labels: map[string]string{
-				"api":       "clusterapi",
-				"apiserver": "true",
-			},
-		},
-		Data: map[string][]byte{
-			"tls.crt": []byte(tlscrt),
-			"tls.key": []byte(tlskey),
-		},
+	caKeyPair, err := triple.NewCA(fmt.Sprintf("%s-certificate-authority", name))
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create root-ca: %v", err)
 	}
-}
 
-const (
-	cabundle = `-----BEGIN CERTIFICATE-----
-MIIC9DCCAdygAwIBAgIBADANBgkqhkiG9w0BAQsFADArMSkwJwYDVQQDEyBjbHVz
-dGVyYXBpLWNlcnRpZmljYXRlLWF1dGhvcml0eTAeFw0xODA4MDkxNDQxNTVaFw0y
-ODA4MDYxNDQxNTVaMCsxKTAnBgNVBAMTIGNsdXN0ZXJhcGktY2VydGlmaWNhdGUt
-YXV0aG9yaXR5MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA5u7BGB85
-+3OOCHWCBB6+XSRNCqzXSk0BPZ9b0lnGAdeK4wqshvIHagCvXtFdLwCms1PWo3G0
-T3a1chzMPRPSpWMmrWCz/Hnms8n9FEcf/ADbWrl8Wb9jBelseXiPKgraV8vzz7LG
-rot6LWd4F5TiJ8Canhiwu1jOTd+ab4tqsFCSNBd4sMw4cihz3cW6h6vfNRmgeCiG
-5zxKdQNPMOO+gMkB6IQvOpnLvYD1lCNDLLv4ssJ6w1mLeTvF0OAOp95lnOy41KIJ
-V+TZTiLdvyhWQ2bDzL/loo6+GIBy5bLxgAWqTpd7nU94IsXJnCSPGm3S5N//zN2R
-ysM55ryR6ULZZQIDAQABoyMwITAOBgNVHQ8BAf8EBAMCAqQwDwYDVR0TAQH/BAUw
-AwEB/zANBgkqhkiG9w0BAQsFAAOCAQEA2Br70cLOPW3X7lzJupe8ZAMC8RN+1KpN
-1/1v6+R5Mq0KgJQsNqgP/YAdTtaoqTlhea2qLzpOH0f4yMekarHOweqf5K/Rp/7h
-Mo4ocOVTfmw9+Vffx6OQTxqM6uvK3IwzfPIkna1alKzANiqVC9Q844Ls0n6D2Ck5
-+n9Ro6MGywX2nEoP7vlRGvpwz11WFcqcOMjwY5uiIiuIR8hS6jNJbz9H/0Ng50wz
-NHRNsymZGoLKX00An2rUVy0NwNYr400GEQBuppKl4rZ6l5PTx7fh7/3wAN3JfXE0
-P66QocO5ZrXAqwezokmymjx31khIgFvNl8Tc1WhEf7MOmdSwWnQCPw==
------END CERTIFICATE-----`
-)
+	apiServerKeyPair, err := triple.NewServerKeyPair(
```

Comment:
The certs need to be generated everytime `ClusterAPIServerAPIServiceObjects` is called since it's expected the cluster API is can be deployed in any namespace. Given the certs are namespace parametrized when generated.